packages/core/src/components/private/BaseInputRange/index.tsx

```diff
@@ -0,0 +1,205 @@
+import React from 'react';
+import memoize from 'lodash/memoize';
+
+import withStyles, { WithStylesProps } from '../../../composers/withStyles';
+import Text from '../../Text';
+import { stylesheetInputRange, HANDLE_SIZE, HALF_HANDLE_SIZE, MARK_SIZE } from './styles';
+
+/**
+ * By default, the **edges** (not the _center_) of the slider handle align with
+ * the **edges** of the slider track. Thus the total width of the slider is not
+ * `100%` but `100% - HANDLE_SIZE_PX`. The default position is correct at 50%,
+ * so we offset based on a fractional distance from the center ()
+ * position based on the percent
+ */
+const getCorrectedPosition = memoize(
```

Comment:
could css's `calc()` have helped with calculating `100% - HANDLE_SIZE_PX`? so it could still be % based?

Reply:
I can play around with that. I think we would still potentially need the `px` calculations for knowing when to hide the annotations when they overlap with the handle (I couldn't figure out zIndex of the handle, maybe can play with that more too, now that everything else is working)

Reply:
@stefhatcher I'm curious, this `calc` still wouldn't truly be `%` based since it would be a `px` offset. Are you suggesting this because it should be more performant than the current implementation?

Reply:
true - just a suggested if it helped clear up any of the calculation logic! could be more performant putting it on the browser, but not sure